lib/team_api/joiner.rb

```diff
@@ -75,7 +75,18 @@ def join_project_data
       # convention takes hold, hopefully.
       projects = (data['projects'] ||= {})
       projects.delete_if { |_, p| p['status'] == 'Hold' } if @public_mode
-      projects.values.each { |p| join_team_list p['team'] }
+      projects.values.each do |p| 
+        if p['errors'].nil?
+          p['errors'] = []
+        end
+        join_team_list p['team'], p['errors']
+        if !p['team'].nil?
+          p['team'].compact!
+        end
+        if p['errors'].empty?
```

Comment:
Another alternative for this block: `p.delete 'errors' unless p['errors'].empty?`.